collect_app/src/main/java/org/odk/collect/android/utilities/LaunchState.kt

```diff
@@ -0,0 +1,34 @@
+package org.odk.collect.android.utilities
+
+import android.content.Context
+import androidx.preference.PreferenceManager
+import org.odk.collect.android.preferences.keys.MetaKeys
+import org.odk.collect.shared.Settings
+import java.io.File
+
+class LaunchState(
+    private val context: Context,
+    private val metaSettings: Settings,
+    private val currentVersion: Int
+) {
+
+    fun isUpgradedFirstLaunch(): Boolean {
+        return if (metaSettings.contains(MetaKeys.LAST_LAUNCHED)) {
+            metaSettings.getInt(MetaKeys.LAST_LAUNCHED) < currentVersion
+        } else {
+            val legacyMetadataDir = File(context.getExternalFilesDir(null), "metadata")
```

Comment:
So this checks if it's upgraded from version without projects and above all the future cases? If so it would be good to comment it somehow as well or create two separate methods with descriptive names.

Reply:
I put a comment on the tests but I think you're right that a descriptive method would be nice here.